### PR TITLE
hyfetch: init at 1.1.2

### PR DESCRIPTION
**Pull request by YisuiDenghua**
###### Description of changes
Added [HyFetch](https://github.com/hykilpikonna/hyfetch) package.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/hyfetch/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ python3Packages, lib, fetchFromGitHub }:
```
Comment (viraptor) [inline pkgs/tools/misc/hyfetch/default.nix:1]: The policy is `lib` first, then sort by usage order.
```suggestion
{ lib, python3Packages, fetchFromGitHub }:
```

File: pkgs/tools/misc/hyfetch/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  };
+
+  propagatedBuildInputs = with python3Packages; [
+    typing-extensions
+    setuptools
+  ];
+
+  doCheck = false;
+
+  meta = with lib; {
```
Comment (viraptor) [inline pkgs/tools/misc/hyfetch/default.nix:21]: No maintainers?

File: pkgs/tools/misc/hyfetch/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    rev = "92623417f90f0cf006c0dd2adcf3f24f4308fe0c";
+    sha256 = "sha256-26L2qt+RarRf3+L6+mMy/ZJNVBVirKs5oEclEsImtC0=";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
+    typing-extensions
+    setuptools
+  ];
+
+  doCheck = false;
```
Comment (viraptor) [inline pkgs/tools/misc/hyfetch/default.nix:19]: The tests pass if you remove the numpy-importing one. Not a huge deal in this case, but it's possible to correct if you want.

File: pkgs/tools/misc/hyfetch/default.nix (hunk 2 above)
Comment (YisuiDenghua) [reply pkgs/tools/misc/hyfetch/default.nix:21]: Added myself as the maintainer.

**Review by viraptor — COMMENTED** [review]
Result of `nixpkgs-review pr 182944` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hyfetch</li>
  </ul>
</details>

You should rebase the PR into two commits: adding the maintainer and adding the package.
But otherwise all 👍

File: pkgs/tools/misc/hyfetch/default.nix (hunk 3 above)
Comment (YisuiDenghua) [reply pkgs/tools/misc/hyfetch/default.nix]: I added `preCheck` to remove the file, thank you for reminding me.

File: pkgs/tools/misc/hyfetch/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      operating system or distribution you are running, what theme or
+      icon set you are using, etc.
+    '';
+    homepage = "https://github.com/hykilpikonna/HyFetch";
+    license = licenses.mit;
+    mainProgram = "hyfetch";
+    maintainers = [
+      maintainers.yisuidenghua
+    ];
+    platforms = platforms.all;
```
[the remark this answers is not in the document]
Comment (YisuiDenghua) [reply pkgs/tools/misc/hyfetch/default.nix]: done. thanks

**Review by viraptor — APPROVED** [review]

**Review by stigtsp — APPROVED** [review]
LGTM, thx for the contribution! <3

**Pull request merged**